io/genwqe/genwqetest.py

```diff
@@ -0,0 +1,203 @@
+        """
+        data_tar = "data.tar.gz"
+        shutil.copyfile(self.test_tar, data_tar)
+        self.files_used.append(data_tar)
+        data_dir = "/usr/share/testdata/"
+        self.dirs_used.append(data_dir)
+        if not os.path.isdir(data_dir):
+            os.makedirs(data_dir)
+        cmd = "genwqe_test_gz -A SW -C %s -v -t %s" % (self.card, data_tar)
+        if process.system(cmd, shell=True, ignore_status=True):
+            self.fail("genwqe_test_gz SW fails")
+        cmd = "genwqe_test_gz -A GENWQE -S -C %s -v -t %s" % (self.card,
+                                                              data_tar)
+        if process.system(cmd, shell=True, ignore_status=True):
+            self.fail("genwqe_test_gz SW forced fails")
+        cmd = "genwqe_test_gz -A GENWQE -C %s -v -t %s" % (self.card, data_tar)
+        if process.system(cmd, shell=True, ignore_status=True):
+            self.fail("genwqe_test_gz GENWQE fails")
+
+    def test_genwqe_memcopy(self):
+        """
+        Tests genwqe_memcopy on the device.
+        """
+        failures = []
+        in_file = "in.bin"
+        out_file = "out.bin"
+        self.files_used.extend([in_file, out_file])
+        for err in ['', '0x1', '0x2', '0x4', '0x8']:
+            cmd = "dd if=/dev/urandom bs=4096 count=1000 of=%s" % in_file
+            process.system(cmd, shell=True)
+            err_opt = ''
+            if err:
+                err_opt = " -Y %s" % err
+            cmd = "genwqe_memcopy -A GENWQE -C %s -c 1000 -v -F -D %s " \
+                "--patternfile %s %s" % (self.card, err_opt, in_file, out_file)
+            if process.system(cmd, shell=True, ignore_status=True) and not err:
+                self.fail("genwqe_memcopy %s fails" % err_opt)
+            cmd = "diff %s %s" % (in_file, out_file)
+            if process.system(cmd, shell=True, ignore_status=True):
+                failures.append(err_opt)
+            for fil in [in_file, out_file]:
+                os.remove(fil)
+        if failures:
+            self.fail("%s fails on file comparison" % ", ".join(failures))
+
+    def test_genwqe_gzip_gunzip(self):
+        """
+        Tests genwqe_gzip and genwqe_gunzip on the device.
+        """
+        self.files_used.extend(["file", "file.gz"])
+        cmd = "dd if=/dev/urandom bs=4096 count=100000 of=file"
+        process.system(cmd, shell=True)
+        for opt in ['', '-1', '-9']:
+            orig_size = os.stat("file").st_size
+            cmd = "genwqe_gzip -A GENWQE -B %s file %s" % (self.card, opt)
+            if process.system(cmd, shell=True, ignore_status=True):
+                self.fail("genwqe_gzip %s fails" % opt)
+            cmd = "genwqe_gunzip -A GENWQE -B %s file.gz %s" % (self.card, opt)
+            if process.system(cmd, shell=True, ignore_status=True):
+                self.fail("genwqe_gunzip %s fails" % opt)
+            processed_size = os.stat("file").st_size
+            if orig_size != processed_size:
```

Comment:
yup i already saw that i think both place can be have with this change again it is just a suggestion and it's your call , please let me know what you think